python/mxnet/model.py

```diff
@@ -848,7 +848,7 @@ def fit(self, X, y=None, eval_data=None, eval_metric='acc',
         # init optmizer
         if isinstance(self.optimizer, str):
             batch_size = data.batch_size
-            if kvstore and 'dist' in kvstore.type and not '_async' in kvstore.type:
+            if kvstore and 'dist' in kvstore.type and '_async' not in kvstore.type:
                 batch_size *= kvstore.num_workers
```

Comment:
is this unit tested to validate it still works after these changes? if so, where?

Reply:
need unit test for both cases (in and not in)

Reply:
I didn't change the logic, this is logically equivalent and was forced by pylint. Feel free to ask @mli, it's his code. I think that request is out of scope for this PR.

Reply:
let's not use veto for this purpose if possible

Reply:
I'm happy to explain why NOT(TRUE) is equivalent to FALSE.

Reply:
why did lint never catch this before?
how can we be sure this has the same effect?

Reply:
> why did lint never catch this before?
Previously, we had two pylint installs in place. One with a pinned version and at a later point in the execution chain, one which installs the latest version. The latter has been removed.

> how can we be sure this has the same effect?
The code change? Try it out yourself: 
```if (not 'test' in 'test') is ('test' not in 'test'):
    print("equivalent)
else:
    print("not equivalent")```

Reply:
you’re making a change on something that so far as I know has been passing pylint already. I am asking if it is unit tested. Also, why did it pass before and not now?

Reply:
I’ve seen large systems broken due to what is perceived as a “trivial change” on more than one occasion.

Reply:
@szha open questions are not vetos.

Reply:
Pylint started to complain just now, the reason is stated above. 

We don't have the tools to check whether this exact code path is unit tested. I didn't really bother to search for it since this change is logically equivalent.

I have supplied you with code to prove that the outcome is identical. Please prove proof that my reformatting is not equivalent. The actions of the underlying code are not part of this PR and thus out of scope.

Reply:
ok

Reply:
Approved?

Reply:
@cjolivier01 you asked other committers to respect your "request changes" reviews as vetos before, hence I assumed that your "request changes" review in this occasion should be considered veto too. Should I change my assumption? If so, what's the criteria, so that we could avoid miscommunication. If not, my request stands.

Reply:
@szha I’m not sure where you’re going with that comment.  Veto is an Apache concept to block a code merge.  If you know of a “don’t merge until either questions are answered sufficiently or code changes are made” button, then please point it out.  In the past, unanswered questions have led to last-minute fiascos (ie MKL changes PR).

Reply:
As far as I understand, we're using the request changes button as a veto - non veto requests are with the comment function, equivalent to +1,0,-1. In accordance with the Apache guidelines, a veto might be ignored if it is outdated or heavily against the interest of the project.

Reply:
@cjolivier01 I'm referring to your comment in https://github.com/apache/incubator-mxnet/pull/9931#issuecomment-370880967
![image](https://user-images.githubusercontent.com/2626883/37691613-50a43b6c-2c70-11e8-9c76-a742bb64daca.png)

Reply:
Then came the following:
![image](https://user-images.githubusercontent.com/2626883/37691818-d447912a-2c71-11e8-8bc7-3a19f63fa1ef.png)
![image](https://user-images.githubusercontent.com/2626883/37691824-e7f89dd6-2c71-11e8-993b-d696fada61f8.png)
Clearly there are conflicting messages, so I think it would be better to clarify them.

Reply:
@szha see email there where you personally pointed out your view that committer questions need not always be answered, thus leaving no choice but to block with a request changes until such time that requester makes the changes or makes sufficient arguments against them (or in this case shows they are not needed).

5th paragraph of second email from top:

https://lists.apache.org/thread.html/5f7843a77f58239dbf317f8fc811be1560d0f6602514adc5487811c6@%3Cdev.mxnet.apache.org%3E

Reply:
@cjolivier01 I'm merely trying to understand what your take is on that, given that you expressed that you didn't intend to veto while still made "request changes" review.

Reply:
So what's the verdict?